platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1046,6 +1055,9 @@ - (id)mgl_jsonExpressionObject {
             
         case NSFunctionExpressionType: {
             NSString *function = self.function;
+
+            BOOL hasCollectionProperty = !(self.arguments.firstObject.expressionType == NSConstantValueExpressionType || self.arguments.firstObject.expressionType == NSKeyPathExpressionType
```

Comment:
The precondition for this check is the member of `self.arguments` is NSExpression. This requires the same thing I mentioned that if we create NSExpression with `expressionForFunction`, the arguments must be NSExpression. Otherwise it will crash if the elements are not NSExpressions.